app/org/sagebionetworks/bridge/validators/SurveyAnswerValidator.java

```diff
@@ -176,45 +171,62 @@ private void validateType(Errors errors, DecimalConstraints con, String answer)
 
     private void validateType(Errors errors, IntegerConstraints con, String answer) {
         try {
-            int value = Integer.parseInt(answer);
-            if (con.getMinValue() != null && value < con.getMinValue()) {
-                rejectField(errors, "constraints", "%s is lower than the minimum value of %s", answer, con.getMinValue());
-            }
-            if (con.getMaxValue() != null && value > con.getMaxValue()) {
-                rejectField(errors, "constraints", "%s is higher than the maximum value of %s", answer, con.getMaxValue());
-            }
-            if (con.getStep() != null && value % con.getStep() != 0) {
-                rejectField(errors, "constraints", "%s is not a step value of %s", answer, con.getStep());
-            }
+            validateNumericalConstraint(errors, con, answer);
         } catch(NumberFormatException e) {
             rejectField(errors, "constraints", "%s is not a valid integer", answer);
         }
     }
 
-    private void validateType(Errors errors, StringConstraints con, String answer) {
-        if (con.getMinLength() != null && answer.length() < con.getMinLength()) {
-            rejectField(errors, "constraints", "%s is shorter than %s characters", answer, con.getMinLength());
-        } else if (con.getMaxLength() != null && answer.length() > con.getMaxLength()) {
-            rejectField(errors, "constraints", "%s is longer than %s characters", answer, con.getMaxLength());
-        }
-        if (StringUtils.isNotBlank(con.getPattern()) && answer != null && !answer.matches(con.getPattern())) {
-            rejectField(errors, "constraints", "%s does not match the regular expression /%s/", answer, con.getPattern());
-        }
-    }
-
     private void validateType(Errors errors, DurationConstraints con, String answer) {
         if (con.getUnit() == null) {
             rejectField(errors, "constraints", "%s does not have a specified unit", answer);
+        } else if (!Unit.DURATION_UNITS.contains(con.getUnit())) {
+            rejectField(errors, "constraints", "%s is not a time unit", answer);
```

Comment:
Yes, most of what you're looking for is in SurveyAnswerValidatorTest... but I'll double-check for each one.